internal/config/from_file.go

```diff
@@ -0,0 +1,322 @@
+		if nodes[i].Value == "hosts" {
+			hostsData.Content = append(hostsData.Content, nodes[i+1].Content...)
+		} else {
+			mainData.Content = append(mainData.Content, nodes[i], nodes[i+1])
+		}
+	}
+
+	mainBytes, err := yaml.Marshal(&mainData)
+	if err != nil {
+		return err
+	}
+
+	filename := ConfigFile()
+	err = WriteConfigFile(filename, yamlNormalize(mainBytes))
+	if err != nil {
+		return err
+	}
+
+	hostsBytes, err := yaml.Marshal(&hostsData)
+	if err != nil {
+		return err
+	}
+
+	return WriteConfigFile(HostsConfigFile(), yamlNormalize(hostsBytes))
+}
+
+func (c *fileConfig) Aliases() (*AliasConfig, error) {
+	// The complexity here is for dealing with either a missing or empty aliases key. It's something
+	// we'll likely want for other config sections at some point.
+	entry, err := c.FindEntry("aliases")
+	var nfe *NotFoundError
+	notFound := errors.As(err, &nfe)
+	if err != nil && !notFound {
+		return nil, err
+	}
+
+	toInsert := []*yaml.Node{}
+
+	keyNode := entry.KeyNode
+	valueNode := entry.ValueNode
+
+	if keyNode == nil {
+		keyNode = &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Value: "aliases",
+		}
+		toInsert = append(toInsert, keyNode)
+	}
+
+	if valueNode == nil || valueNode.Kind != yaml.MappingNode {
+		valueNode = &yaml.Node{
+			Kind:  yaml.MappingNode,
+			Value: "",
+		}
+		toInsert = append(toInsert, valueNode)
+	}
+
+	if len(toInsert) > 0 {
+		newContent := []*yaml.Node{}
+		if notFound {
+			newContent = append(c.Root().Content, keyNode, valueNode)
+		} else {
+			for i := 0; i < len(c.Root().Content); i++ {
+				if i == entry.Index {
+					newContent = append(newContent, keyNode, valueNode)
+					i++
+				} else {
+					newContent = append(newContent, c.Root().Content[i])
+				}
+			}
+		}
+		c.Root().Content = newContent
+	}
+
+	return &AliasConfig{
+		Parent:    c,
+		ConfigMap: ConfigMap{Root: valueNode},
+	}, nil
+}
+
+func (c *fileConfig) hostEntries() ([]*HostConfig, error) {
+	entry, err := c.FindEntry("hosts")
+	if err != nil {
+		return nil, fmt.Errorf("could not find hosts config: %w", err)
+	}
+
+	hostConfigs, err := c.parseHosts(entry.ValueNode)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse hosts config: %w", err)
+	}
+
+	return hostConfigs, nil
+}
+
+// Hosts returns a list of all known hostnames configured in hosts.yml
+func (c *fileConfig) Hosts() ([]string, error) {
+	entries, err := c.hostEntries()
+	if err != nil {
+		return nil, err
+	}
+
+	hostnames := []string{}
+	for _, entry := range entries {
+		hostnames = append(hostnames, entry.Host)
+	}
+
+	sort.SliceStable(hostnames, func(i, j int) bool { return hostnames[i] == ghinstance.Default() })
+
+	return hostnames, nil
+}
+
+func (c *fileConfig) DefaultHost() (string, error) {
+	val, _, err := c.DefaultHostWithSource()
+	return val, err
+}
+
+func (c *fileConfig) DefaultHostWithSource() (string, string, error) {
+	hosts, err := c.Hosts()
+	if err != nil {
+		return ghinstance.Default(), "", nil
+	}
+
+	if len(hosts) == 1 {
```

Comment:
These two conditions could be combined to simplify the expression:
```suggestion
	if err == nil && len(hosts) == 1 {
```